src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    // if (typeof separator !== 'string') {
+    //   return this;
+    // }
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = `${this[0]}`;
+
+    if (this[0] === null) {
+      result = '';
+    }
+
+    if (this.length === 1) {
```

Comment:
This condition is also not needed because the loop in line 24 will not execute if 'this.length' is 1.